### PR TITLE
FIX: 'read' filter in top menu showing new topics

### PR DESCRIPTION
**Pull request by vinothkannans**
https://meta.discourse.org/t/read-filter-in-top-menu-showing-new-topics/61654

### CODE REVIEW

File: lib/topic_query.rb [hunk 1]
```diff
@@ -165,7 +165,8 @@ def list_latest
 
   def list_read
     create_list(:read, unordered: true) do |topics|
-      topics.order('COALESCE(tu.last_visited_at, topics.bumped_at) DESC')
+      topics.where('tu.last_visited_at IS NOT NULL')
+        .order('COALESCE(tu.last_visited_at, topics.bumped_at) DESC')
```
Comment (eviltrout) [inline lib/topic_query.rb]: If `tu.last_visited_at IS NOT NULL` then the `COALESCE` here isn't doing anything and can be removed.

I would also add a spec to make sure it works properly.

Comment (vinothkannans) [inline lib/topic_query.rb]: But I can see the change in result. Normally it includes new topics. After i added the `where` condition it showing correctly.

Anyway adding spec will give best solution for this 👍

Comment (vinothkannans) [inline lib/topic_query.rb]: Sorry. I misunderstood. Now only I read about what is `COALESCE`. I will update this PR soon.

**Comment by eviltrout** [comment]
Thanks for removing the `COALESCE` - it still needs a test though.

**Comment by vinothkannans** [comment]
@eviltrout I am trying to add. Not yet found any example. I will add it asap :+1:.

**Comment by eviltrout** [comment]
It looks like this is the existing spec for `list_read`:

https://github.com/discourse/discourse/blob/master/spec/components/topic_query_spec.rb#L459-L464

Perhaps you could just add a new topic in the `it` block and then make sure it's not included?

**Comment by vinothkannans** [comment]
@eviltrout it's done :+1:

**Comment by eviltrout** [comment]
Thanks, looks great!

**Pull request merged**